src/HotChocolate/Language/src/Language.SyntaxTree/ObjectFieldNode.cs

```diff
@@ -85,7 +84,7 @@ public bool Equals(ObjectFieldNode? other)
             return true;
         }
 
-        return other.Name.Equals(Name) && other.Value.Equals(Value);
+        return other.Name.IsEqualTo(Name) && other.Value.IsEqualTo(Value);
```

Comment:
in this case Equals is correct. IsEqualsTo is a helper for nullable props for an object field the name can never be null. The same goes for value. In the syntax tree, the value can never be null.